Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1

```diff
@@ -0,0 +1,356 @@
+    Write-Host "Cloud service(s) with disabled RDP access: [$($totalCloudServiceWithDisabledRDPAccess)]"
+    Write-Host "`t"
+
+    # Performing remediation on cloud service(s).
+    try
+    {
+        if ($totalCloudServiceWithEnableRDPAccess -gt 0)
+        {
+            # Creating the log file
+            Write-Host "Backing up config of cloud service(s) detail." -ForegroundColor $([Constants]::MessageType.Info)
+            $cloudServiceWithEnabledRDPAccess | ConvertTo-json | out-file "$($folderpath)\CloudServiceWithEnabledRDPAccess.json"  
+            Write-Host "Path: $($folderpath)\CloudServiceWithEnabledRDPAccess.json"     
+            
+            # Asking user to verify logs and select 'Y' to proceed
+            if(-not $Force)
+            {
+                Write-Host "Do you want to disable RDP access from cloud service(s) listed in above path?" -ForegroundColor Yellow -NoNewline
+                $UserInput = Read-Host -Prompt "(Y|N)"
+                if($UserInput -ne "Y")
+                {
+                    return;
+                }
+            }
+            
+            Write-Host "`t"
+            Write-Host "Disabling RDP access on [$($totalCloudServiceWithEnableRDPAccess)] cloud service(s)..."
+            try 
+            {
+                $cloudServiceWithEnabledRDPAccess | ForEach-Object {
+                    $serviceName = $_.CloudServiceName
+                    $rgName = $_.ResourceGroupName
+                    
+                    # Disabling RDP access
+                    $_.RDPExtensionDetails | ForEach-Object {
+                        $output = Remove-AzureServiceRemoteDesktopExtension -ServiceName $serviceName -Slot $_.Slot -Role $_.Role -ErrorAction SilentlyContinue
+                        if($null -eq $output -and $output.OperationStatus -ine "Succeeded")
+                        {
+                            $item =  New-Object psobject -Property @{  
+                                CloudServiceName = $serviceName                
+                                ResourceGroupName = $rgName
+                            }
+    
+                            $skippedCloudServiceFromRemediation += $item
+                            break;
+                        }
+                    }
+
+                    if(($skippedCloudServiceFromRemediation | Measure-Object).Count -eq 0)
+                    {
+                        $_ | Select-Object @{Expression={($_.ResourceGroupName)};Label="ResourceGroupName"},@{Expression={$_.CloudServiceName};Label="CloudServiceName"} | Sort-Object | Format-Table | Out-String
+                    }
+                }
+            }
+            catch
+            {
+                $item =  New-Object psobject -Property @{  
+                    CloudServiceName = $_.CloudServiceName                
+                    ResourceGroupName = $_.ResourceGroupName
```

Comment:
Do you want to include the extension details here?

Reply:
Here we are showing service and RG name of cloud services for which we are unable to do remediation(due to permission issue or any command failure).
Therefore showing extension details is not required